# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:20px;">**Web Scraping**</p>

<p align="center">
  <img src="https://hasdata.com/assets/cache_image/components/images/blog/web-scraping.14c36f6b.jpg" alt="Web Scraping" width="600">
  <br>
  Source : Hasdata.com
</p>


**Author  : Salsa Sabitha Hurriyah**  

**Program ini dibuat untuk Menganalisis penjualan seblak di platform Tokopedia.**  

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:20px;">**1. Introduction**</p>

## Business Understanding/Problem Statement

Problem : Ingin menambah Income dengan berjualan seblak, namun tidak punya cukup modal. Sehingga memutuskan untuk menjalankan skema dropship di platform Tokopedia. Namun masih ragu apakah seblak adalah produk yang tepat untuk dijual.  

**S** (Spesific) : Untuk meyakinkan apakah seblak adalah produk yang tepat untuk dijual agar dapat menambah Income dari berjualan seblak, harus mengetahui potensi penjualan seblak. Maka akan dilakukan analisis terhadap penjualan produk seblak di platform Tokopedia untuk memahami tingkat minat dan respons masyarakat terhadap seblak.   
    
**M** (Measurable) : Mendapatkan penambahan Income dengan ekspektasi tambahan sebesar Rp 3 juta per bulannya.  
   
**A** (Achievable) : Menganalisis penjualan seblak di Tokopedia.  

**R** (Relevant) : Analisis ini relevan dengan tujuan untuk melihat potensi penjualan seblak yang dapat menghilangkan keraguan dalam pengambilan keputusan untuk menjual seblak atau tidak. Apabila seblak memiliki potensi yang bagus, maka dengan berjualan seblak dapat menambah income per bulannya.  
    
**T** (Time Bond) : Hasil solusi dari analisis ini bisa dilihat dalam waktu 1 bulan kedepan setelah proses analisa data selesai.  
   
Meningkatkan pendapatan dari berjualan seblak dengan target tambahan pendapatan sebesar Rp 3 juta perbulannya. Dilakukan dengan menganalisis penjualan produk seblak di platform Tokopedia. Tujuannya adalah untuk memahami tingkat minat dan respons masyarakat terhadap seblak. Analisis yang dilakukan ini relevan untuk menghilangkan keraguan dalam mengambil keputusan untuk berjualan seblak. Hasil analisis ini akan dievaluasi dalam waktu satu bulan setelah selesai dilakukan analisis data.

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:20px;">**2. Import Libraries**</p>

In [2]:
# Import libraries
import pandas as pd
import numpy as np
from scipy import  stats
from selenium import webdriver
from bs4 import BeautifulSoup
import time 

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:20px;">**3. Web Scraping**</p>

In [2]:
# Assign driver
driver = webdriver.Chrome()

### Data yang diambil :  
- Nama Produk  
- Harga Produk  
- Penjual  
- Kota Toko  
- Banyaknya Terjual  
- Rating Produk  
  
Ambil informasi produk minimal dari 10 halaman pencarian dengan jumlah data yang diperoleh minimal 50

Membuat looping untuk meng-scraping 10 halaman web Tokopedia

In [106]:
# Membuat list kosong untuk menyimpan list data yang diminta
list_nama_produk = []
list_harga_produk = []
list_nama_toko = []
list_kota_toko = []
list_banyak_terjual = []
list_rating_produk = []

# Looping untuk 10 halaman
for i in range (1,11):
    # Mengakses website
    url = "https://www.tokopedia.com/search?navsource=&page={}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st=".format(i)
    driver.get(url)

    # meyimpan html menjadi variable
    html = driver.page_source
    soup = BeautifulSoup(html, 'html.parser')
    time.sleep(5)
    
    # Medefinisikan element box
    boxes = soup.find_all('div', {'class':'css-54k5sq'})

    # Memubat variabel untuk menyimpan list data per halaman
    nama_produk = []
    harga_produk = []
    nama_toko = []
    kota_toko = []
    banyak_terjual = []
    rating_produk = []
    
    for box in boxes:
        # Mengambil data nama produk
        try:
            nama = box.find('div', {'class':'prd_link-product-name css-3um8ox'})
            nama_produk.append(nama.get_text())
        except:
            nama_produk.append(None)

        # Mengambil data harga produk
        try:
            harga = box.find('div', {'class':'prd_link-product-price css-h66vau'})
            harga_produk.append(harga.get_text())
        except:
            harga_produk.append(None)

        # Mengambil data penjual
        penjual = box.find('span', {'class':"prd_link-shop-name css-1kdc32b flip"})
        try:
            nama_toko.append(penjual.get_text())
        except:
            nama_toko.append(None)

        # Mengambil data kota toko
        kota = box.find('span', {'class':"prd_link-shop-loc css-1kdc32b flip"})
        try:
            kota_toko.append(kota.get_text())
        except:
            kota_toko.append(None)

        # Mengambil data banyak terjual
        terjual = box.find('span', {'class':'prd_label-integrity css-1sgek4h'})
        try:
            banyak_terjual.append(terjual.get_text())
        except:
            banyak_terjual.append(None)

        # Mengambil data rating
        rating = box.find('span', {'class':'prd_rating-average-text css-t70v7i'})
        try:
            rating_produk.append(rating.get_text())
        except:
            rating_produk.append(None)

    # Menambahkan list data
    list_nama_produk += nama_produk
    list_harga_produk += harga_produk
    list_nama_toko += nama_toko
    list_kota_toko += kota_toko
    list_banyak_terjual += banyak_terjual
    list_rating_produk += rating_produk

Insight : Looping berhasil dilakukan dan data telah disimpan dalam list masing-masing kolom

In [117]:
# Mengecek panjang data
print(len(list_nama_produk))
print(len(list_harga_produk))
print(len(list_nama_toko))
print(len(list_kota_toko))
print(len(list_banyak_terjual))
print(len(list_rating_produk))

1065
1065
1065
1065
1065
1065


Insight : Panjang data tiap kolom sama, yaitu 1065


### Menyimpan hasil scrape ke pandas data frame dan kemudian diolah

In [204]:
df = pd.DataFrame({'nama_produk' : list_nama_produk,
                   'harga_produk' : list_harga_produk,
                   'toko' : list_nama_toko,
                   'kota' : list_kota_toko,
                   'terjual' : list_banyak_terjual,
                   'rating' : list_rating_produk})
df.to_csv('Data_Seblak.csv', index=False)

Insight  :  Data hasil scrape disimpan ke dalam data frame pandas dengan variabel df dan disimpan ke file csv dengan nama file Data_Seblak

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:20px;">**4. Data Preparation**</p>

In [3]:
# Variabel baru untuk membaca file csv yang sudah disimpan (agar data asli tidak berubah)
df1 = pd.read_csv('Data_Seblak.csv')

### 1. Lakukan eksplorasi data sederhana:  

- Menampilkan beberapa baris data

In [6]:
df1.head()

nama_produk harga_produk  \
0  Seblak Instan Ceu Nthien Khas Bandung Rasana N...     Rp17.000   
1  GFS Seblak Basah Instan Original - Tinggal Sed...     Rp16.618   
2  KERUPUK MENTAH KERUPUK SEBLAK MENTAH 500 GR - ...     Rp12.500   
3                               Seblak instant murah      Rp4.900   
4  Kerupuk Seblak Pedas Krupuk Pelangi Pedas Sebl...     Rp15.500   

                       toko               kota       terjual  rating  
0  Central Seblak Nusantara  Tangerang Selatan  2rb+ terjual     4.9  
1      Lakoca Official Shop             Cimahi  100+ terjual     4.9  
2                BociKakang    Jakarta Selatan  100+ terjual     4.9  
3               dapurbulbin            Bandung           NaN     NaN  
4                 toko bemo      Jakarta Timur  500+ terjual     4.9

Insight : Data pada data frame harus dilakukan cleaning agar data dapat diproses

- Menampilkan informasi rangkuman data

In [7]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1065 entries, 0 to 1064
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   nama_produk   350 non-null    object 
 1   harga_produk  350 non-null    object 
 2   toko          350 non-null    object 
 3   kota          350 non-null    object 
 4   terjual       344 non-null    object 
 5   rating        343 non-null    float64
dtypes: float64(1), object(5)
memory usage: 50.1+ KB


Insight : Terdapat 6 kolom dan 1065 baris

- Cek missing value

In [8]:
df1.isnull().sum()

nama_produk     715
harga_produk    715
toko            715
kota            715
terjual         721
rating          722
dtype: int64

Insight : Terdapat 715 missing value pada kolom nama_produk, harga_produk, toko, dan kota. sedangkan untuk kolom terjual terdapat 721 missing value, sedangkan untuk kolom rating terdapat 722 missing value.

- Cek Duplicate

In [9]:
df1['nama_produk'].duplicated().sum()

959

Insight : Terdapat 959 data duplikat

### 2. Lakukan data cleaning

Handle missing value :

In [4]:
df1 = df1.dropna(subset=['kota'])
df1.isnull().sum()

nama_produk     0
harga_produk    0
toko            0
kota            0
terjual         6
rating          7
dtype: int64

Insight : Menghapus missing value pada kolom kota

Handle missing value pada kolom terjual 

In [5]:
df1 = df1.replace(to_replace=[None], value='0')
df1.isnull().sum()

nama_produk     0
harga_produk    0
toko            0
kota            0
terjual         0
rating          7
dtype: int64

Insight : Tersisa 7 missing value pada kolom rating

Handle missing value pada kolom rating

In [6]:
df1 = df1.replace(to_replace=[np.nan], value=0)
df1.isnull().sum()

nama_produk     0
harga_produk    0
toko            0
kota            0
terjual         0
rating          0
dtype: int64

Insight : Sudah tidak terdapat missing value

Menghapus data nama produk yang duplikat :

In [7]:
df1 = df1.drop_duplicates(subset=['nama_produk'], keep='first')
df1.duplicated().sum()

0

Insight : Sudah tidak terdapat data duplikat

Me-replace karakter yang tidak diperlukan :

In [8]:
# Menghilangkan Rp pada kolom harga_produk
df1['harga_produk'] = df1['harga_produk'].str.replace('Rp', '')
# Menghilangkan + pada kolom terjual
df1['terjual'] = df1['terjual'].str.replace('+', '')
# Menghilangkan 'terjual' pada kolom terjual
df1['terjual'] = df1['terjual'].str.replace('terjual', '')
# Mengganti rb menjadi 000 pada kolom terjual
df1['terjual'] = df1['terjual'].str.replace('rb', '000')

df1

nama_produk harga_produk  \
0    Seblak Instan Ceu Nthien Khas Bandung Rasana N...       17.000   
1    GFS Seblak Basah Instan Original - Tinggal Sed...       16.618   
2    KERUPUK MENTAH KERUPUK SEBLAK MENTAH 500 GR - ...       12.500   
3                                 Seblak instant murah        4.900   
4    Kerupuk Seblak Pedas Krupuk Pelangi Pedas Sebl...       15.500   
..                                                 ...          ...   
960  Sebring seblak Rafael bumb pedas gurih daun je...        4.951   
961  KERUPUK SEBLAK 500GR / KRUPUK SEBLAK DAUN JERU...       19.900   
962           Seblak Makroni pedas daun jeruk 500 gram       25.000   
963  Kerupuk Bawang Warna Putih Dan Orange 500Gram ...       11.000   
964          seblak Abah aceng paling top pedas mantul        5.000   

                         toko               kota terjual  rating  
0    Central Seblak Nusantara  Tangerang Selatan   2000      4.9  
1        Lakoca Official Shop             Cimahi    100      4.9  
2                  BociKakang    Jakarta Selatan    100      4.9  
3                 dapurbulbin            Bandung       0     0.0  
4                   toko bemo      Jakarta Timur    500      4.9  
..                        ...                ...     ...     ...  
960          Grosir Putra Bdg       Kab. Bandung     25      5.0  
961             Dapur Sehat21       Kab. Bandung     80      4.9  
962      Seblak putra bandung       Kab. Bandung     14      5.0  
963          Distributor-Jaya          Tangerang    100      4.9  
964      cemilan Bandungstore             Cimahi     23      5.0  

[105 rows x 6 columns]

Insight : Data sudah bersih, kemudian akan dilakukan penyesuaian tipe data

Cek summary data

In [20]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 105 entries, 0 to 964
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   nama_produk   105 non-null    object 
 1   harga_produk  105 non-null    object 
 2   toko          105 non-null    object 
 3   kota          105 non-null    object 
 4   terjual       105 non-null    object 
 5   rating        105 non-null    float64
dtypes: float64(1), object(5)
memory usage: 5.7+ KB


Insight : Kolom yang tipe datanya tidak sesuai yaitu harga produk yang seharusnya float dan terjual yang seharusnya integer

Menyesuaikan tipe data :

In [9]:
# Mengganti tipe data pada kolom harga_produk menjadi float
df1['harga_produk'] = df1['harga_produk'].astype(float)

# Mengganti tipe data pada kolom terjual menjadi integer
df1['terjual'] = df1['terjual'].astype(int)

df1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 105 entries, 0 to 964
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   nama_produk   105 non-null    object 
 1   harga_produk  105 non-null    float64
 2   toko          105 non-null    object 
 3   kota          105 non-null    object 
 4   terjual       105 non-null    int32  
 5   rating        105 non-null    float64
dtypes: float64(2), int32(1), object(3)
memory usage: 5.3+ KB


Insight : Tipe data tiap kolom sudah sesuai dan siap untuk digunakan di tahap selanjutnya

In [10]:
# Menyimpan tabel data seblak yang sudah di cleaning
df1.to_csv('Data_Seblak_Cleaned.csv', index=False)

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:20px;">**5. Analisis**</p>

Untuk menganalisis penjualan seblak, langkah pertama yang dilakukan yaitu :   

Menghitung rata-rata, median, standar deviasi, skewness, dan kurtosis dari kolom :
- Harga  
- Banyak produk terjual  
- Rating

In [3]:
# Membaca file csv tabel data seblak yang sudah di cleaning
df2 = pd.read_csv('Data_Seblak_Cleaned.csv')

# Membuat tabel statistik
table_stat = pd.DataFrame(df2[['harga_produk','terjual','rating']].describe()).T

# Membuat tabel median
median = pd.DataFrame([df2[['harga_produk','terjual','rating']].median()])
median = median.rename(index={0:'median'})

# Membuat tabel mean
mean = pd.DataFrame([df2[['harga_produk','terjual','rating']].mean()])
mean = mean.rename(index={0:'mean'})

# Membuat tabel standar deviasi
std = pd.DataFrame([df2[['harga_produk','terjual','rating']].std()])
std = std.rename(index={0:'std'})

# Membuat series skewness dan kurtosis
skewness = df2[['harga_produk','terjual','rating']].skew()
kurtosis = df2[['harga_produk','terjual','rating']].kurtosis()

# Membuat dataframe normTestFrame berdasarkan skewness dan kurtosis dan mengubah indexnya menjadi skewness dan kurtosis
distributionTest = pd.DataFrame([skewness,kurtosis])
distributionTest = distributionTest.rename(index={0:'skewness',1:'kurtosis'})

# Menggabungkan tabel median, mean, dan distributionTest
table_stat = pd.concat([median,mean,std,distributionTest],ignore_index=False)
table_stat

harga_produk      terjual     rating
median       16.000000   100.000000   4.900000
mean         29.438381   721.742857   4.608571
std          72.308646  1717.978620   1.147834
skewness      8.071966     3.880059  -3.787718
kurtosis     72.631398    16.720349  12.801481

Insight :
- Dilihat dari median dan mean, kolom harga_produk dan terjual memiliki perbedaan nilai yang cukup signifikan. sedangkan kolom rating memiliki perbedaan nilai yang tidak terlalu signifikan.
- Dilihat dari nilai standar deviasi, kolom harga_produk dan terjual memiliki standar deviasi yang tinggi, sedangkan rating tidak
- Dilihat dari skewness harga_produk dan terjual memiliki skewness positif yang tinggi, dan rating memiliki skewness negatif yang juga tinggi
- Dilihat dari kurtosis harga_produk, terjual, dan rating memiliki kurtosis > 0 (leptokurtic).

Diketahui bahwa distribusi data sangat tidak merata dan data memiliki outlier

### Selanjutnya akan dilakukan Test Normalitas
Jika p-value > 0.05 data terdistribusi normal.

- D’Agostino’s K^2 Test

In [5]:
hargatest = stats.normaltest(df2['harga_produk']).pvalue
terjualtest = stats.normaltest(df2['terjual']).pvalue
ratingtest = stats.normaltest(df2['rating']).pvalue

print(f'p-value harga produk : {hargatest}')
print(f'p-value terjual : {terjualtest}')
print(f'p-value rating : {ratingtest}')

p-value harga produk : 9.227403500146866e-44
p-value terjual : 2.0630667329351847e-25
p-value rating : 5.503004942159435e-24


Insight :
Dalam D'Agostinos test semua data memiliki distribusi normal

- Shapiro-Wilk Test

In [6]:
hargatest2 = stats.shapiro(df2['harga_produk']).pvalue
terjualtest2 = stats.shapiro(df2['terjual']).pvalue
ratingtest2 = stats.shapiro(df2['rating']).pvalue

print(f'p-value shapiro harga produk : {hargatest2}')
print(f'p-value shapiro terjual : {terjualtest2}')
print(f'p-value shapiro rating : {ratingtest2}')

p-value shapiro harga produk : 1.5683884020108125e-20
p-value shapiro terjual : 5.08673311482443e-18
p-value shapiro rating : 7.712469180261334e-20


Insight :
Dalam Shapiro-Wilk test semua data memiliki distribusi normal

#### Selanjutnya untuk mengetahui potensi pendapatan dari berjualan seblak 
Dilakukan dengan menghitung lower value dan upper value dari distribusi populasi pendapatan dimana confidence levelnya adalah 95%.

In [7]:
pendapatan = df2['terjual'] * df2['harga_produk']
std = pendapatan.std()
N = len(df2)
low, up = stats.norm.interval(0.95,loc=pendapatan.mean(),scale=std/np.sqrt(N))
print('Lower :', low)
print('Upper :', up)

Lower : 5599.322317667436
Upper : 22654.38415852304


Insight : Estimasi pendapatan terendah tiap bulannya adalah Rp5.599.000 dan tertinggi Rp22.654.000

#### Mendeteksi outliers :  
Karena skewness data > 0.5 maka pencarian outlier dilakukan dengan menggunakan Tukey Rule

Tukey Rule :

In [8]:
# Mengambil data quartil 1 dan 3 dari tabel_stat
q1 = table_stat.quantile(0.25)
q3 = table_stat.quantile(0.75)
iqr = q3-q1

# Mencari lower Bound dan Upper Bound
upperBound = q3 + 1.5*iqr
lowerBound = q3 - 1.5*iqr

# Mencari Outlier
outlier_harga = df2['harga_produk'][(df2['harga_produk'] < lowerBound['harga_produk']) | (df2['harga_produk'] > upperBound['harga_produk'])].copy()
outlier_terjual = df2['terjual'][(df2['terjual'] < lowerBound['terjual']) | (df2['terjual'] > upperBound['terjual'])].copy()
outlier_rating = df2['rating'][(df2['rating'] < lowerBound['rating']) | (df2['rating'] > upperBound['rating'])].copy()

persen_harga = len(outlier_harga) * 100/len(df2['harga_produk'])
persen_terjual = len(outlier_terjual) * 100/len(df2['terjual']) 
persen_rating = len(outlier_rating) * 100/len(df2['rating']) 

print(f'Banyak outlier harga_produk : {len(outlier_harga)}')
print(f'Banyak outlier terjual : {len(outlier_terjual)}')
print(f'Banyak outlier rating : {len(outlier_rating)}')
print()
print(f'Persentase outlier harga_produk : {persen_harga}%')
print(f'Persentase outlier terjual : {persen_terjual}%')
print(f'Persentase outlier rating : {persen_rating}%')

Banyak outlier harga_produk : 2
Banyak outlier terjual : 13
Banyak outlier rating : 0

Persentase outlier harga_produk : 1.9047619047619047%
Persentase outlier terjual : 12.380952380952381%
Persentase outlier rating : 0.0%


Insight :  
Diketahui bahwa terjual memiliki outlier tertinggi diikuti dengan harga_produk

### Setelah mendeteksi outlier, akan dilakukan Pembersihan Outlier
Dengan mengambil data yang selain outlier

In [9]:
# Mengambil data quartil 1 dan 3 dari tabel_stat
q1 = table_stat.quantile(0.25)
q3 = table_stat.quantile(0.75)
iqr = q3-q1

# Mencari lower Bound dan Upper Bound
upperBound = q3 + 1.5*iqr
lowerBound = q3 - 1.5*iqr

# Mencari Outlier
harga_bersih =  df2.query('harga_produk >= %s and harga_produk <= %s'%(lowerBound['harga_produk'],upperBound['harga_produk'])).copy()
terjual_bersih = df2.query('terjual >= %s and terjual <= %s'%(lowerBound['terjual'],upperBound['terjual'])).copy()
rating_bersih = df2.query('rating >= %s and rating <= %s'%(lowerBound['rating'],upperBound['rating'])).copy()

persen_harga_bersih = len(harga_bersih) * 100/len(df2['harga_produk'])
persen_terjual_bersih = len(terjual_bersih) * 100/len(df2['terjual']) 
persen_rating_bersih = len(rating_bersih) * 100/len(df2['rating']) 

print(f'Banyak data tanpa outlier harga_produk : {len(harga_bersih)}')
print(f'Banyak data tanpa outlier terjual : {len(terjual_bersih)}')
print(f'Banyak data tanpa outlier rating : {len(rating_bersih)}')
print()
print(f'Persentase tanpa outlier harga_produk : {persen_harga_bersih}%')
print(f'Persentase tanpa outlier terjual : {persen_terjual_bersih}%')
print(f'Persentase tanpa outlier rating : {persen_rating_bersih}%')

Banyak data tanpa outlier harga_produk : 103
Banyak data tanpa outlier terjual : 92
Banyak data tanpa outlier rating : 105

Persentase tanpa outlier harga_produk : 98.0952380952381%
Persentase tanpa outlier terjual : 87.61904761904762%
Persentase tanpa outlier rating : 100.0%


Insight : 
Diketahui bahwa rating tidak memiliki outlier,  
Data telah dibersihkan dari outlier yang kemudian siap untuk diproses untuk tahap selanjutnya.

### Terdapat beberapa kota lokasi toko, akan dilakukan perbandingan harga di Jabodetabek dan di luar Jabodetabek  
Dengan perbedaan biaya bahan baku di kedua lokasi, maka akan dibuat hipotesis untuk mengetahui pakah harga barang di Jabodetabek dan di luar Jabodetabek berbeda.  

#### Langkah pertama yaitu Pembuatan Hipotesis  
Ada kemungkinan harga seblak di jabodetabek dan di luar jabodetabek berbeda dikarenakan perbedaan harga bahan baku

Hipotesis :  
- H0 : Tidak ada perbedaan harga seblak di Jabodetabek dan di Luar Jabodetabek
- H1 : Harga seblak di Jabodetabek dan di Luar Jabodetabek Berbeda

#### Langkah selanjutnya yaitu Pengujian Hipotesis  
Metode pengujian yang digunakan :  
- Pengujian dengan membandingkan rata-rata
- Pengujian Two Sample T-Test

Membuat data frame untuk menyimpan data Jabodetabek dan luar Jabodetabek

In [58]:
# Membuat list variabel untuk menampung kota yg termasuk Jabodetabek
jabodetabek = ['jakarta','bogor','depok','tangerang','bekasi']

# Membuat list variabel kosong untuk menyimpan nilai harga di Jabodetabek dan Luar Jabodetabek
hargaIn = []
hargaOut = []

# Looping untuk setiap baris data frame
for index, row in df2.iterrows():
    # Mengubah format kolom kota menjadi lower agar dapat diidentifikasi dengan variabel baru
    kota_lower = row['kota'].lower()
    # Jika ada setidaknya satu lokasi dalam jabodetabek yang terdapat dalam kota_lower, maka kondisi tersebut akan terpenuhi (True).
    if any(lok in kota_lower for lok in jabodetabek):
        # Maka akan menambahkan baris harga dan kota ke dalam variabel hargaIn
        hargaIn.append(row['harga_produk'])
    else:
        # Jika tidak terdapat dalam jabodetabek maka akan menambahkan baris harga dan kota tersebut ke dalam variabel hargaOut
        hargaOut.append(row['harga_produk'])

# Menjadikan hargaIn dan hargaOut menjadi data frame dan diubah nama kolom nya
hargaIn = pd.DataFrame(hargaIn).rename(columns={0:'harga'})
hargaOut = pd.DataFrame(hargaOut).rename(columns={0:'harga'})

# Menampilkan banyak toko dari jabodetabek dan luar jabodetabek
print(f'Banyak toko Jabodetabek : {len(hargaIn)}')
print(f'Banyak toko luar Jabodetabek : {len(hargaOut)}')

Banyak toko Jabodetabek : 63
Banyak toko luar Jabodetabek : 42


Insight : Diketahui bahwa lebih banyak toko yang berasal dari jabodetabek dibandingkan luar jabodetabek

#### Selanjutnya Pengujian Hipotesis :  
-  Pengujian dengan membandingkan rata-rata

In [60]:
# Membuat variabel untuk menampung nilai mean
meanIn = hargaIn['harga'].mean()
minIn = hargaOut['harga'].min()
maxIn = hargaOut['harga'].max()
stdIn = hargaIn['harga'].std()
meanOut = hargaOut['harga'].mean()
minOut = hargaOut['harga'].min()
maxOut = hargaOut['harga'].max()
stdOut = hargaOut['harga'].std()

# Membuat variabel untuk hipotesis
H0 =  meanIn == meanOut
H1 =  meanIn != meanOut

# Menampilkan nilai mean
print(f'Rata-rata harga produk di Jabodetabek : {meanIn}')
print(f'Harga terendah produk di Jabodetabek : {minIn}')
print(f'Harga tertinggi produk di Jabodetabek : {maxIn}')
print(f'Standar deviasi harga produk di Jabodetabek : {stdIn}')
print()
print(f'Rata-rata harga produk luar Jabodetabek : {meanOut}')
print(f'Harga terendah produk di Jabodetabek : {minOut}')
print(f'Harga tertinggi produk di Jabodetabek : {maxOut}')
print(f'Standar deviasi harga produk luar Jabodetabek : {stdOut}')

Rata-rata harga produk di Jabodetabek : 29.026174603174606
Harga terendah produk di Jabodetabek : 3.5
Harga tertinggi produk di Jabodetabek : 240.672
Standar deviasi harga produk di Jabodetabek : 88.17223536307013

Rata-rata harga produk luar Jabodetabek : 30.05669047619047
Harga terendah produk di Jabodetabek : 3.5
Harga tertinggi produk di Jabodetabek : 240.672
Standar deviasi harga produk luar Jabodetabek : 38.802863985628996


Insight : Diketahui bahwa harga rata-rata Jabodetabek dan Luar Jabodetabek tidak ada perbedaan yang signifikan.

- Pengujian Hipotesis dengan Two Sample Test  

In [74]:
t_stat, p_val = stats.ttest_ind(hargaIn['harga'],hargaOut['harga'])
print('T-Statistic:',t_stat)
print('P-value:',p_val) 

T-Statistic: -0.07119956794677094
P-value: 0.9433769890990458


Insight :  
- Dari nilai T-Statistik dapat diketahui bahwa tidak ada perbedaan signifikan antara harga seblak jabodetabek dan luar jabodetabek, dengan harga jabodetabek lebih rendah dari pada luar jabodetabek.  
- Nilai P-Value > 0.05 yang artinya gagal menolak H0 (H0 Diterima)

Kesimpulan : H0 diterima. Tidak ada perbedaan harga antara Jabodetabek dan Luar Jabodetabek

### Untuk mengetahui apakah orang lebih cenderung suka dengan produk yang harganya murah?  
Pembuatan Hipotesis :  
H0 = Tidak ada korelasi antara harga dan rating  
H1 = Ada korelasi antara harga dan rating

Uji korelasi yang digunakan :  
Karena rating merupakan data bertipe kategorikal sedangkan harga merupakan numerikal, maka uji korelasi yang digunakan adalah Kendall.

Uji Korelasi Kendall antara harga dan rating

In [8]:
# Menghitung nilai korelasi dan p-value dari kolom harga_produk dan rating
corr_tau, pval_k = stats.kendalltau(df2['harga_produk'], df2['rating'])

# Menampilkan nilai korelasi dan p-value dari kolom harga_produk dan rating
print(f'P-Value Kendall : {pval_k}')
print(f'Nilai korelasi kendall : {corr_tau}')

# jika p-value kendall besar dari 0.05
if pval_k > 0.05:
    # maka menampilkan Korelasi kedua variabel bersifat kebetulan
    print('Korelasi kedua variabel bersifat kebetulan')
else:
    # Jika tidak maka akan menampilkan Korelasi kedua variabel bersifat nyata
    print('Korelasi kedua variabel bersifat nyata')

P-Value Kendall : 0.04631557413028574
Nilai korelasi kendall : 0.14764283940561812
Korelasi kedua variabel bersifat nyata


Insight : P-value < 0,05 maka korelasi kedua variabel bersifat nyata. H0 ditolak

Kesimpulan : H0 ditolak. Dapat disimpulkan bahwa murah mahalnya harga sangat mempengaruhi rating.

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:20px;">**6. Kesimpulan**</p>

Kesimpulan yang dapat diambil dari hasil analisis ini adalah :  

- Dari 10 halaman yang di scrap, diperoleh 1065 data. Setelah dilakukan cleaning data diperoleh data sebanyak 105 baris.
- Potensi pendapatan penjualan seblak di Platform Tokopedia berkisar dari Rp5.599.000 sampai Rp22.654.000 perbulan.
- Untuk wilayah Jabodetabek dan Luar Jabodetabek tidak memiliki perbedaan harga yang signifikan.
- Harga produk mempengaruhi rating pembeli.  

**Berdasarkan dari analisis ini, saya akan mengambil keputusan untuk menjadi dropshipper seblak karena potensi pesimis keuntungan sebesar Rp5.599.000 per bulannya dimana pendapatan tersebut sudah memenuhi target saya yaitu sebesar Rp 3 juta perbulannya**